internal/metrics/prometheus/prometheus.go

```diff
@@ -0,0 +1,179 @@
+/**
+ * Copyright (c) F5, Inc.
+ *
+ * This source code is licensed under the Apache License, Version 2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+package prometheus
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"log/slog"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/nginx/agent/v3/internal/bus"
+	"github.com/nginx/agent/v3/internal/metrics"
+)
+
+const (
+	defaultTarget  = "http://127.0.0.1:30882/metrics"
+	DataSourceType = "PROMETHEUS"
+)
+
+//go:generate go run github.com/maxbrunsfeld/counterfeiter/v6@v6.7.0 -generate
+//counterfeiter:generate -o mock_message_bus.go . MessageBusContract
+//go:generate sh -c "grep -v github.com/nginx/agent/v3/internal/metrics/prometheus mock_message_bus.go | sed -e s\\/prometheus\\\\.\\/\\/g > mock_message_bus_fixed.go"
+//go:generate mv mock_message_bus_fixed.go mock_message_bus.go
+type MessageBusContract interface {
+	Register(int, []bus.Plugin) error
+	DeRegister(plugins []string) error
+	Process(...*bus.Message)
+	Run()
+	Context() context.Context
+	GetPlugins() []bus.Plugin
+	IsPluginAlreadyRegistered(string) bool
+}
+
+// Used for deserializion of parsed Prometheus data.
+type Scraper struct {
+	bus bus.MessagePipeInterface
```

Comment:
not sure if we should be passing the bus in here. I think the bus should only be used by plugins and not passed to other places in the codebase.

Reply:
Yeah, this will be removed after we change the data source contract as you suggested [here](https://github.com/nginx/agent/pull/544#discussion_r1469848053).

Reply:
The MessagePipe is now only used by the Metrics Plugin, so this is now resolved.